src/ncp/ncp_host.cpp

```diff
@@ -71,6 +75,50 @@ void NcpHost::GetDeviceRole(DeviceRoleHandler aHandler)
     mNcpSpinel.GetDeviceRole(aHandler);
 }
 
+void NcpHost::Join(const otOperationalDatasetTlvs &aActiveOpDatasetTlvs, const AsyncResultReceiver aReceiver)
+{
+    ChainedAsyncResultPtr result;
+    bool                  isAlready    = !mJoinReceivers.empty();
+    auto                  errorHandler = [this](otError aError, const std::string &aErrorInfo) {
+        CallAsyncReceivers(mJoinReceivers, aError, aErrorInfo);
+    };
+
+    mJoinReceivers.push_back(aReceiver);
```

Comment:
What if this call is using a different operational dataset from another ongoing call?

Reply:
Good point! What would be the expected behavior on Android? 

Under current implementation, the different operational dataset won't take effect. But it will still get callbacked when previous operation completes.

I think we may probably change to let each call of 'Join' just go through the process. Sometimes if the spinel has already started an operation, then returns `BUSY`. Thoughts?

Reply:
Have changed the implementation to allow a consequent call (even previous one is not allowed). Both calls will be get callbacked anyway.